manager/testdata/new-blueprints/copy-read-blueprint.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: app.m4d.ibm.com/v1alpha2
+kind: Blueprint
+metadata:
+  name: copy-and-read
+  labels:
+spec:
+  cluster: thegreendragon
+  appSelector: # Selector of the application that uses this workload
+    clusterName: thegreendragon
+    workloadSelector:
+      matchLabels:
+        app: demoapp
+  flow:
+  - name: copy-step
+    assetID: 'm4d-notebook-sample/paysim-csv'
+    templateSpec:
+      chart: ghcr.io/mesh-for-data/m4d-implicit-copy-batch:0.1.0
+    moduleLabel: "mod: demoapp-copy-step"
+    frequency: "0 1 * * *" # How often should the process be repeated (this is crontab syntax but could also be something "like once per day")
+    source:
+      vault:
+        address: http://vault.m4d-system:8200
+        authPath: /v1/auth/kubernetes/login
+        role: module
+        secretPath: /v1/kubernetes-secrets/paysim-csv?namespace=m4d-notebook-sample
+      format: parquet
+      connection:
+        s3:
+          endpoint: localhost:8001
+          bucket: srcbucket
+          object: data.parq
+    destination:
+      vault:
+        address: http://vault.m4d-system:8200
+        authPath: /v1/auth/kubernetes/login
+        role: module
+        secretPath: /v1/kubernetes-secrets/paysim-csv?namespace=m4d-notebook-sample
+      format: parquet
+      connection:
+        s3:
+          endpoint: localhost:8001
+          bucket: destbucket
+          object: data.parq
+    actions:
+    - action: redact
+      column: blood_group
+  - name: read-step
+    templateSpec:
+      chart: ghcr.io/mesh-for-data/m4d-arrow-flight:0.1.0
+    moduleSelector: "mod: demoapp-read-step"
+    assetID: 'm4d-notebook-sample/paysim-csv'
+    source:
+      step: copy-step # the name of step for debugging purposes
```

Comment:
Can you please explain again how the read module would know if the bucket in the source is ready? `step: copy-step` is the release name of the dependent copy module which the current read module could investigate for its status but I think you had another solution in mind. Thanks.

Reply:
The read module does not need to know if its source is ready. The read module is ready when the arrow flight server is up and running. The copy is marked ready when the data is copied. Only when both modules are ready the plotter as well as the application is marked as ready. If a user wants to access a resource before it's marked as ready there may of course be errors. The point is that the readyness check get's pushed to the overall ready status of the application. (Only when all modules that are involved in an asset are successful the asset is also ready)

Reply:
Thanks for the explanation! that makes sense. so we assume no module will have to check another module's status or this is a specific case? Thanks